### PR TITLE
RSDK-7796 add gantry nil checks

### PR DESCRIPTION
**Pull request by oliviamiller**
tested with rands fake module and the empty gantry RC card now works.

### CODE REVIEW

File: components/gantry/server.go [hunk 1]
```diff
@@ -53,6 +59,11 @@ func (s *serviceServer) GetLengths(
 	if err != nil {
 		return nil, err
 	}
+
+	// if the value returned is nil, return an empty array.
+	if lengthsMm == nil {
+		lengthsMm = []float64{}
+	}
```
Comment (martha-johnston) [inline components/gantry/server.go:66]: for this and the position check above, wondering if there is actually a case in which these are returned as nil? just quickly looked at `lengths` in the actual gantry driver and it doesn't seem so but I could it on the server/client side?

Comment (oliviamiller) [reply components/gantry/server.go:66]: yeah none of our builtins return nil but a modular gantry could

Comment (martha-johnston) [reply components/gantry/server.go:66]: ah good point

Comment (martha-johnston) [reply components/gantry/server.go:66]: but wouldn't it have to match the function signature of returning `([]float64, error)`? I'm not super familiar with the module rules so maybe this isn't enforced

Comment (oliviamiller) [reply components/gantry/server.go:66]: Array and pointer return values can be nil. If i change to just `float64 ` in the module it doesn't let me return nil so i think it just depends on golang rules if the return type is allowed to be nil or not.

**Review by martha-johnston — APPROVED** [review]

**Pull request merged**